backend/server.js

```diff
@@ -228,6 +228,64 @@ app.get('/api/classes', async (req, res) => {
   }
 })
 
+// Create or Edit Class
+app.post('/api/classes', async (req, res) => {
+  try {
+    const { title, level, ageGroup, instructor, schedule } = req.body;
+
+    // Validate required fields
+    if (!title || !level || !ageGroup || !instructor || !schedule) {
+      return res.status(400).json({ message: 'Missing required fields' });
+    }
+
+    // Check if class already exists by title (assuming title is unique)
+    const existingClass = await Class.findOne({ title });
+
+    if (existingClass) {
+      // If class exists, update it while preserving the roster
+      const updatedClass = await Class.findOneAndUpdate(
+        { title },
```

Comment:
Rather than use the title to filter for the class to update, it's better to use the id of the existing class, that way we can also update the title.